### DISCUSSION TITLE
Official Fork?

### DISCUSSION THREAD

**Discussion by mrfroggg** [jertel/elastalert2#77] [Ideas]
Qmando doesn't give any update and he has been saying, for many months, that he'll soon come back with an official strategy to transfer the project ton another maintainer.
Users are still asking questions / opening issues in the main Elastalert repo, probably most of them don't know about this "alternate" EA as it's hard to find if you are not looking for it.

On my side, some people around me are starting to question if we need to replace EA because of the lack of development on the official repo.

So, is it time to do an official fork? I think we should not wait anymore for Qmando/Yelp, from what I understood, they are not using Elasticsearch anymore.
@jertel and @nsano-rururu have been working hard on EA!

Fork name could be: elastalert2 with a 1.0 release!

**Comment by ferozsalam** [comment]
I am in favour of this as well, this project has regained some momentum with this fork and I think it would be a shame not to make the most of it.

My only note would be that having the name 'Elastalert 2' with a 1.0 release number could be slightly confusing. Perhaps something like 'Elastalert Neo' or similar to indicate it's a different project but not tying it to a particular number? In any case this is only a minor detail.

**Reply by mrfroggg** [reply]
yeah, I found out that @jertel created a pypi release of "elastalert2" with version 2.0.2 (February release), so it might be the way to go

**Comment by Shashankft9** [comment]
I observed that both @jertel and  @nsano-rururu have different forks?
https://github.com/jertel/elastalert
https://github.com/nsano-rururu/elastalert

Perhaps we can get just one new source of truth first? And have the same people as maintainers?

**Reply by nsano-rururu** [reply]
@Shashankft9

Do you allow forking and developing? .. You can judge for yourself.

**Reply by nsano-rururu** [reply]
@Shashankft9

I'm just forking to investigate bugs. Or do you request that the forked repository be erased?

**Reply by nsano-rururu** [reply]
@Shashankft9

I am the maintainer of Praeco and Elast Alert Server below. jertel / elastalert is making pull requests for mistakes and bugs in the ElastAlert documentation found during development.
https://github.com/johnsusek/praeco
https://github.com/johnsusek/elastalert-server

**Reply by Shashankft9** [reply]
I am saying that it would probably be better if there is one aligned fork which the new users can look up to. I have noticed that you have been putting considerable amount of work into your fork and sometimes in this fork too, which is nice actually. But I wish that there is just one fork that everyone focuses on. 
For example, I have been doing some changes in my own fork but its private and I'd like to put it into a public fork, like yours or this one, but I would be confused on which public fork to contribute? :(

**Reply by mrfroggg** [reply]
@nsano-rururu I think he is saying that your current fork and @jertel's fork are not at the same level on the commit sides.

You two seem to be the most active developers (thanks!), so if we want to have an official fork outside of Yelp/elastalert / Qmando, we'll have to be able to merge all commits together and work on the same repo to have the same base code.

**Reply by nsano-rururu** [reply]
@Shashankft9

Then tell everyone.
https://github.com/Yelp/elastalert/network/members

**Reply by nsano-rururu** [reply]
@mrfroggg @Shashankft9

I'm not an official folk guy, though. It's a problem if you think that you are a related person, so please understand that.

**Reply by nsano-rururu** [reply]
@mrfroggg

Don't give my name. Because it doesn't matter

**Reply by mrfroggg** [reply]
Ok so, if @jertel is ok with it, he could transform this repo to an official fork and add you as a maintainer, so you could merge your commit (and future ones) from your fork.

In the Todo for @jertel and/or someone else:
- Add an "Issues" tab on this repo
- Name new maintainers / collaborators
- Update readthedocs (I think it's already done on https://elastalert2.readthedocs.io/ )
- Create official pypi releases (https://pypi.org/project/elastalert2/ ?)
- Spread the word on Yelp/elastalert that a new fork as been created

**Reply by ferozsalam** [reply]
I am happy to help with the administrative side of things if @jertel is interested in doing this and wants assistance.

**Reply by jertel** [reply]
Good discussion here. I'm on board with this and appreciate your offers to help maintain the repository.

As far as making this the 'official' fork, we have to assume that such a declaration by the Yelp team won't occur, which means this fork will be no different from any of the other thousands of Elastalert forks on Github (as @nsano-rururu correctly pointed out), except that we'll have more momentum behind us from the additional work we've already put into this particular fork. We can do some things to try and keep that momentum flowing, such as posting notices to the Discussions, Issues, and PR tabs of the Yelp/Elastalert repository. But it will take a good amount of time for someone to stumble across our fork on a Google search, for instance as we'll not be showing up on the the first page of search results for some time.

Regarding the changes need, thank you @mrfroggg for starting that list. Is that list what you had in mind to 'transform this repo to an official fork'? If so I can get started on that over the next week. Or is there some additional tweaks you were expecting for the repository itself?

My original thought earlier this year was to keep the Elastalert name since it's already familiar, and is just a great name for what this project is, but then append a '2' to better distinguish this project as being the next generation of the original Elastalert. If we're ok with this then I'm interested in whether you all would like this repository renamed to _elastalert2_. 

Also, if we proceed, I would like to merge the related _jertel/elastalert-docker_ project into this one, to simplify maintenance and builds. I would then make a new _elastalert2_ repository on Docker Hub and deprecate the existing elastalert-docker repository on Docker Hub.

Let me know your thoughts.

**Reply by ferozsalam** [reply]
I think that all sounds reasonable!

Given the number (and the low quality) of most of the issues raised on the Yelp Elastalert repo, I think we should pre-emptively consider how issues should be handled, and ensure that we're only keeping issues around if they are valid and we are planning to fix them. I know that it's sometimes seen as a controversial, but something like a [stale bot](https://github.com/marketplace/stale) might be the best way of keeping noise down.

**Reply by mrfroggg** [reply]
I agree with @ferozsalam , valid issues should only be kept, call for help/support should be done elsewhere.
In my previous work contract, I worked a lot with Netbox (https://github.com/netbox-community/netbox)  and the repo admin was pretty strict and it worked very well, issues are super clean! They also had the stale bot.
They have a Google Group for discussion / support.
On Yelp/elastalert, they have a chat-kind of thing for support, but I have to say that I never found anything useful in there, we should propose something a bit more useful.

@jertel : 'transform this repo to an official fork', yes, it's what I had in mind: issues, documentation, release! Feel free to start on it! I can create a Google Group or something else for support requests.

As for the '2', I agree for Elastalert2!
As an example, I use iTerm2 everyday on my Macs, it replaced the original iTerm. iTerm2 is now at version 3.4.5, the "2" is not bound to version 2.X

**Reply by ferozsalam** [reply]
We could probably continue using this Github discussions board for support requests, rather than creating a separate Google Group - it seems to have everything we would need.

**Reply by jertel** [reply]
I agree with the issues problem, and that's why I turned off the Issues feature for this repo when I created it. I felt that I'd rather have people ask for help in Discussions, and if it's a legitimate issue then they or someone else perhaps can issue a PR for it. But if we have multiple maintainers here willing to moderate an issues list, and potentially a stalebot to close out those that have lost interest, that's a good solution too.

I think the Discussions forum will be OK as a place for support requests. Even though it's a new GitHub feature I've been using it with good results on other projects. It's had a very quick adoption rate.

**Reply by Shashankft9** [reply]
+1 for github discussions, would be better to start off with less places so that they can be easily maintained, if later on people feel would like to branch out to google groups, that would also be fine I think.

**Comment by ferozsalam** [comment]
Just seen all the updates, thanks @jertel it's looking good!

**Reply by devinbfergy** [reply]
I am down for @jertel and that fork being the new elastalert2.0

**Comment by anthonylavado** [comment]
@jertel One thing to consider - searching through code with the GitHub search is unavailable for forks. Over at Jellyfin, we had to contact support to request that the fork relationship be "broken" so that repositories we've forked could have full capabilities. It may be worth doing that for your repository. Yes, it does take it away from the "network", but it lets this project live on independently.

**Reply by jertel** [reply]
Thanks for sharing this. I spent a fair amount of time searching GitHub for a way to break the chain and came up empty handed. 

Also, thanks for your contributions on Jellyfin - great project!

**Reply by jertel** [reply]
This repo is no longer attached to the original elastalert repo. Thanks again for the recommendation.
